src/main/java/com/example/java/maven/ticketVendingMachine/CoinStorage.java

```diff
@@ -0,0 +1,86 @@
+package com.example.java.maven.ticketVendingMachine;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class CoinStorage {
+    private final Map<Coin, Integer> coins;
+    private final MessagePrinter messagePrinter;
+
+    public CoinStorage(Map<Coin, Integer> coins, MessagePrinter messagePrinter) {
+        this.coins = coins;
+        this.messagePrinter = messagePrinter;
+    }
+
+    public Map<Coin, Integer> getCoins() {
+        return coins;
+    }
+
+    public void clear() {
+        coins.clear();
+    }
+
+    public void addCoin(Coin coin) {
+        int currentNumberOfCoins = Optional.ofNullable(coins.get(coin)).orElse(0);
+        coins.put(coin, currentNumberOfCoins + 1);
+    }
+
+    public static void moveCoinsFromTempToMainCoinStorage(CoinStorage mainCoinStorage, CoinStorage tempCoinStorage) {
+        for (Coin coin : Coin.values()) {
+            if (tempCoinStorage.getCoins().keySet().contains(coin)) {
+                int numberOfGivenCoinsInTempStorage = getNumberOfCoinsWithGivenValueInGivenCoinStorage(coin.getCoinValue(), tempCoinStorage);
+                int numberOfGivenCoinsInMainStorage = getNumberOfCoinsWithGivenValueInGivenCoinStorage(coin.getCoinValue(), mainCoinStorage);
+                mainCoinStorage.getCoins().put(coin, numberOfGivenCoinsInMainStorage + numberOfGivenCoinsInTempStorage);
+            }
+        }
+        tempCoinStorage.clear();
+    }
+
+    public int getValueOfCoinsInStorage() {
+        int valueOfCoinsInStorage = 0;
+        for (Coin coin : coins.keySet()) {
+            valueOfCoinsInStorage += (coin.getCoinValue() * coins.get(coin));
+        }
+        return valueOfCoinsInStorage;
+    }
+
+    public static int getRequiredNumberOfCoinsWithGivenValue(int oddMoney, int coinValue) {
+        return Math.floorDiv(oddMoney, coinValue);
+    }
+
+    public static boolean areRequiredCoinsAvailableInCoinStorage(int requiredNumberOfCoins, int coinValue, CoinStorage mainCoinStorage) {
+        if (getNumberOfCoinsWithGivenValueInGivenCoinStorage(coinValue, mainCoinStorage) >= requiredNumberOfCoins) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public static int getNumberOfCoinsWithGivenValueInGivenCoinStorage(int coinValue, CoinStorage givenCoinStorage) {
```

Comment:
and even when this method will  be necessary, for sure will not be static

Reply:
Deleted